Lib/test/test_curses.py

```diff
@@ -1107,6 +1107,39 @@ def test_issue6243(self):
         curses.ungetch(1025)
         self.stdscr.getkey()
 
+    def test_in_wch(self):
+        stdscr = self.stdscr
+
+        if not hasattr(stdscr, 'in_wch'):
+            raise unittest.SkipTest('requires curses.window.in_wch')
+
+        if curses.has_colors():
+            curses.init_pair(1, curses.COLOR_RED, 0)
+            expected_pair = 1
+            color_attr = curses.color_pair(expected_pair)
+        else:
+            expected_pair = 0
+            color_attr = 0
+
+        def _norm(ch, attr, color):
+            # for some reason, curses returns some color bits here?
+            return ch, attr & ~curses.A_COLOR, color
+
+        for ch in ('a', '\xe9', '\u20ac', '\U0001f643'):
+            expected = (ch, curses.A_BOLD, expected_pair)
+
+            stdscr.insstr(0, 0, ch, color_attr | curses.A_BOLD)
+            self.assertEqual(_norm(*stdscr.in_wch()), expected)
+
+            stdscr.addstr(0, 0, ch, color_attr | curses.A_BOLD)
+            self.assertEqual(_norm(*stdscr.in_wch(0, 0)), expected)
```

Comment:
Can we use something that is not at (0, 0) as well to check the ordering (y, x)?